src/openvic-simulation/history/DiplomaticHistory.cpp

```diff
@@ -84,61 +103,86 @@ bool DiplomaticHistoryManager::load_diplomacy_history_file(CountryManager const&
 		"alliance", ZERO_OR_MORE, [this, &country_manager](ast::NodeCPtr node) -> bool {
 			Country const* first;
 			Country const* second;
-			Date start, end;
+			Date start;
+			std::optional<Date> end = {};
 
 			bool ret = expect_dictionary_keys(
 				"first", ONE_EXACTLY, expect_identifier_or_string(country_manager.expect_country_str(assign_variable_callback_pointer(first))),
 				"second", ONE_EXACTLY, expect_identifier_or_string(country_manager.expect_country_str(assign_variable_callback_pointer(second))),
 				"start_date", ONE_EXACTLY, expect_identifier_or_string(expect_date_str(assign_variable_callback(start))),
-				"end_date", ONE_EXACTLY, expect_identifier_or_string(expect_date_str(assign_variable_callback(end)))
+				"end_date", ZERO_OR_ONE, expect_identifier_or_string(expect_date_str(assign_variable_callback(end)))
 			)(node);
 
-			alliances.push_back({ first, second, start, end });
+			const Period period = { start, end };
+			alliances.push_back({ first, second, period });
```

Comment:
```suggestion
			alliances.push_back({ first, second, { start, end } });
```
Here you're creating the variable `period` then copying it into the alliance on the next line, if instead you directly create the period object as an argument when constructing the alliance you avoid needing to allocate memory for the intermediate `period` variable and avoid having to do a copy. Same thing applies to all the other cases below.